File: Hash Tables/ContainsDuplicate/ContainsDuplicate.scala
import scala.collection.mutable.Map

object Solution {
    def containsNearbyDuplicate(nums: Array[Int], k: Int): Boolean = {
        var map = Map[Int, Int]()
        
        for (i <- 0 to nums.length -1){
            if (map.contains(nums(i))){
                var diff = (i - map(nums(i))).abs
                if (diff <= k) return true
                else map(nums(i)) = i
            } else {
                map(nums(i)) = i
            }
        }
        
        return false
    }
}

File: Hash Tables/ContainsDuplicate/ContainsDuplicate.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        hash_map = {}
        for i in range(len(nums)):
            if nums[i] in hash_map:
                diff = abs(i - hash_map[nums[i]])
                if diff <= k:
                    return True
                else:
                    hash_map[nums[i]] = i
            else:
                hash_map[nums[i]] = i
        
        return False